tests/e2e_tests/astrom_e2e.py

```diff
@@ -0,0 +1,185 @@
+
+    # NoiseMap
+    with fits.open(fpn_path) as hdulist:
+        fpn_dat = hdulist[0].data
+    with fits.open(cic_path) as hdulist:
+        cic_dat = hdulist[0].data
+    with fits.open(dark_path) as hdulist:
+        dark_dat = hdulist[0].data
+    noise_map_dat_img = np.array([fpn_dat, cic_dat, dark_dat])
+    noise_map_dat = np.zeros((3, detector.detector_areas['SCI']['frame_rows'], detector.detector_areas['SCI']['frame_cols']))
+    rows, cols, r0c0 = detector.unpack_geom('SCI', 'image')
+    noise_map_dat[:, r0c0[0]:r0c0[0]+rows, r0c0[1]:r0c0[1]+cols] = noise_map_dat_img
+    noise_map_noise = np.zeros([1,] + list(noise_map_dat.shape))
+    noise_map_dq = np.zeros(noise_map_dat.shape, dtype=int)
+    err_hdr = fits.Header()
+    err_hdr['BUNIT'] = 'detected EM electrons'
+    ext_hdr['B_O'] = 0
+    ext_hdr['B_O_ERR'] = 0
+    noise_map = data.DetectorNoiseMaps(noise_map_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                                    input_dataset=mock_input_dataset, err=noise_map_noise,
+                                    dq = noise_map_dq, err_hdr=err_hdr)
+    noise_map.save(filedir=astrom_cal_outputdir, filename="mock_detnoisemaps.fits")
+    this_caldb.create_entry(noise_map)
+
+    ## Flat field
+    with fits.open(flat_path) as hdulist:
+        flat_dat = hdulist[0].data
+    flat = data.FlatField(flat_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr, input_dataset=mock_input_dataset)
+    flat.save(filedir=astrom_cal_outputdir, filename="mock_flat.fits")
+    this_caldb.create_entry(flat)
+
+    # bad pixel map
+    with fits.open(bp_path) as hdulist:
+        bp_dat = hdulist[0].data
+    bp_map = data.BadPixelMap(bp_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr, input_dataset=mock_input_dataset)
+    bp_map.save(filedir=astrom_cal_outputdir, filename="mock_bpmap.fits")
+    this_caldb.create_entry(bp_map)
+
+
+    ####### Run the walker on some test_data
+
+    walker.walk_corgidrp(sim_data_filelist, "", astrom_cal_outputdir)
+
+
+    # clean up by removing entry
+    this_caldb.remove_entry(nonlinear_cal)
+    this_caldb.remove_entry(kgain)
+    this_caldb.remove_entry(noise_map)
+    this_caldb.remove_entry(flat)
+    this_caldb.remove_entry(bp_map)
+
+    ## Check against astrom ground truth (target= [], ps=21.8[mas/pixel], na=45)
+    output_files = []
+    for file in os.scandir(astrom_cal_outputdir): # sort between added files and subdirectories
+        if file.is_file():
+            output_files.append(file)
+
+    expected_platescale = 21.8
+    expected_northangle = 45
+    target = (80.553428801, -69.514096821)
+
+    # currently checks each file indiidually
+    ## boresight now takes all frames and averages platescale, northangle, ra, dec
+    ## this now need to check only the one astrometric calibration file
+    astrom_cal = data.AstrometricCalibration(os.path.join(astrom_cal_outputdir, 'AstrometricCalibration.fits'))
+
+    assert astrom_cal.platescale == pytest.approx(expected_platescale, abs=0.5)
```

Comment:
useful to add comments like in the unit test that spell out in words what the threshold is for each value